src/routes/authentication.ts

```diff
@@ -36,9 +36,11 @@ export default async function authentication(fastify: FastifyInstance) {
       chtSession = await ChtSession.create(authInfo, username, password);
     } catch (e: any) {
       console.error(`Login error: ${e}`);
+      const errorMessage = e.message?.includes('permissions') ? e.message : 'Login Failed. Please Try Again';
```

Comment:
Is it always safe to display the message? Would be nice to at least test those scenarios.   If the message is safe, maybe we should just always display it? If the message isn't safe, then this probably isn't a sufficient check ... like what if the username includes `permissions` and is in message?

Reply:
Let me revert to how it was.  The better way would be creating an explicit Error class but that's beyond the scope of this task.

 Let's communicate vaguely to end users with login failed as before. Devs will be able to check the `e.message` safely.

Reply:
https://github.com/medic/cht-user-management/issues/253